runtime/resolver.go

```diff
@@ -46,6 +46,10 @@ type Resolver interface {
 	ResolveInteractive(ctx context.Context) (ResolverResult, error)
 	// ResolveExport resolve data for export (e.g. downloads or reports).
 	ResolveExport(ctx context.Context, w io.Writer, opts *ResolverExportOptions) error
+	// MetricsViewSecurityFields returns the list of mv fields that will be used in the resolver, these can be used by the security engine to enforce field-level security.
+	MetricsViewSecurityFields() []string
+	// SecuredRowFilter returns the row filter for the resolver, which can be used by the security engine to enforce row-level security.
+	SecuredRowFilter() string
```

Comment:
It feels weird to have this logic on the `Resolver` interface since it's not metrics-view specific. Could it be implemented directly in the security engine itself? It currently has different rules for different resource types.

If we wanted to move the security handling of individual resource types out of the security engine, I think the reconcilers might be a more generic way to do it (since there is one reconciler per resource type). But I think it's okay to keep individual resource handling inside the security engine at the moment.

Reply:
If the problem is circular dependencies, consider if e.g. a `metricsviewutil` package might help instead for util functions for metrics view parsing.

Reply:
I also felt this is a little odd. 

The problem is not circular dependency, problem is that for resources like reports/alerts we need to expand transitive access rule to add mv access, field access and row filter rules, however, this information is present in the query args json. Since resolvers (legacy resolver in case of report) already parses properties to create appropriate query and thus has this information, I thought it would be convenient to expose it here otherwise for example for reports will need to parse the properties and use case statement to create appropriate query and get this information.